api/include/opentelemetry/metrics/meter.h

```diff
@@ -36,7 +36,7 @@ class Meter
    * @return a shared pointer to the created Counter.
    */
 
-  virtual nostd::shared_ptr<Counter<long>> CreateLongCounter(
+  virtual nostd::shared_ptr<Counter<int64_t>> CreateLongCounter(
```

Comment:
When "long" is compiled as a 32bit int, this change is an API and ABI breaking change.

This is ok as only metrics are affected, which are not GA yet.

Still, please add a CHANGELOG section to indicate this is a breaking change compared to 1.6.1

Reply:
Do we need to rename the method to `CreateInt64Counter`?
If later we want to support 32bit integers at API level (say for embedded devices where memory is constraint), would it be confusing, or we can just add `CreateShortCounter` for it.

Reply:
thanks, CHANGELOG updated and the function renamed.